### DISCUSSION TITLE
Publishing patch without saving it...

### DISCUSSION THREAD

**Discussion by xynydev** [cables-gl/cables_docs#343] [Feature Requests] [answered]
I just made a patch on cables, and thought it was cool, so I published it. I created a name, it said that it needs tags, so I added tags and during the process pressed the "Save" button multiple times.
So imagine the surprise when my published project is blank!

I **understand** this is _user error_, but;  
I think it would add to the user-friendliness **to be a little more annoying about saving**,  
even if you want to keep the save buttons for different things separate for some reason.

As a first time user who is used to web-apps having autosave and version history built in, this just seems odd to me. 
I find it weird being able to _publish a completed project_ and press _a "Save" button_ multiple times during the process and end up publishing _a completely empty project_ with no way of recovering.

**Comment by pandrr** [comment, marked as answer]
saving the settings does not save the patch.

it will still show the warning that the patch is not saved. 

![Screenshot 2022-09-15 at 11 48 43](https://user-images.githubusercontent.com/98792/190373141-d4116b08-66f0-4dd1-a535-9c38605e2eac.png)

**Reply by xynydev** [reply]
That's not _really_ a warning that the patch is not saved, it's a warning that some changes might not be saved. _I agree, it was a mistake on my part_.
But I do have to point out that a lot of the time after I've already pressed _"Save"_ I specifically ignore those warnings, because I might have accidentally moved something after saving the whole thing as completed.
_What you might get out of this post is that this idiot user didn't save their project, and that's fine._

I'm just trying to point out that _maybe_ **there should be a more clear of a warning when a patch is being published if it hasn't been saved in a long time.**

**Reply by steam0r** [reply]
hm, so what else except for "changes you made may not be saved" should we tell you? especially if you say that you tend to "ignore those warnings"? 

how are we supposed to decide that you "just moved something" vs. "that was an important move"?

**Reply by xynydev** [reply]
@steam0r 
> how are we supposed to decide that you "just moved something" vs. "that was an important move"?

You aren't! That browser warning is good there! That is obviously the user's decision to make most of the time when they're finishing off the session working on a project and that browser warning pops up.
I specifically "tend to ignore those warnings" when I have _just pressed "Save"_, not literally always.

If you read my final and actual point, the only suggestion for a feature I made is this;
**When someone is publishing a project, and hasn't saved their project _in a while_, maybe ask in a dialogue box or something "Do you want to publish your latest saved version of the project from 2 hours ago, or do you want to save this version and then publish?"**

Edit: It'd also be cool if there was a text box in the top right bar, right after the icons, that says when the last time you saved the project is.

**Reply by xynydev** [reply]
and to be clear, i am super sorry for my use of markdown features and capitalized letters, and maybe being somewhat in-between-the-lines angry. this project is actually great.
